lib/ansible/modules/remote_management/hpe/oneview_fc_network.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (2016-2017) Hewlett Packard Enterprise Development LP
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: oneview_fc_network
+short_description: Manage OneView Fibre Channel Network resources.
+description:
+    - Provides an interface to manage Fibre Channel Network resources. Can create, update, and delete.
+version_added: "2.4"
+requirements:
+    - "python >= 2.7.9"
+    - "hpOneView >= 3.1.0"
+author: "Felipe Bulsoni (@fgbulsoni)"
+options:
+    state:
+        description:
+            - Indicates the desired state for the Fibre Channel Network resource.
+              C(present) will ensure data properties are compliant with OneView.
+              C(absent) will remove the resource from OneView, if it exists.
+        choices: ['present', 'absent']
+    data:
+        description:
+            - List with the Fibre Channel Network properties.
+        required: true
```

Comment:
jimi-c said on slack that this is fine.

We have added a new feature in 2.4 that allows specifying the structure of objects inside of a list if you want to do that.  It could also be done as an update to the module.  (I'm not entirely sure how to use it yet, knowledge of usage is still percolating out).